src/htpmd/shared/population_matrix.py

```
@@ -1,18 +1,102 @@
 """
 Implemented class for computing population matrix
 """
+import json
 import os
 import sys
 import os.path
 import numpy as np
 import itertools
+from htpmd.shared.pizza_dump import dump
 sys.path.append(".")
 sys.path.append("../")
 """
-
+Module for methods to compute ion clusters' population matrix either from Pizza package when aggregates.txt is 
+available or from trajectory dump file when otherwise.
 """
 
 
+def compute_pop_matrix_from_pizza_dump(trajectory_reference_path):
+    """
+    Description: Receives the path to simulation folder and computes population matrix using dump pizza_dump module
+
+        This function assumes Li ("90") and N in TFSI ("93") if the meta.json file does not exist
+
+    Args: trajectory_reference_path: path to the trajectory file to be processed (Ex:
+    '0623/poly_9_nion_50_conc_1.5/9-0-485080326-0')
+
+    Returns:
+        pop_matrix: population matrix that includes the number of clusters with different shapes.
+        pop_matrix[i,j] is the number of clusters with i number cations and j number of anions
+    """
+
+    path_to_aggregate_file = os.path.join(trajectory_reference_path, 'aggregates.txt')
+    path_to_meta_file = os.path.join(trajectory_reference_path, 'meta.json')
+
+    # load and read meta.json
+    try:
+        meta_file = open(path_to_meta_file)
+        meta_json = json.load(meta_file)
+
+        # get types for Li and O(TFSI)
+        li_idx = meta_json['cation_raw_type']
+        o_idx = 94  # this information is currently not available in meta.json
+
+    except FileNotFoundError:
+        li_idx = 90
+        o_idx = 94
```

This way of handling a missing file carries a risk for the user, because the program silently uses default values (`li_idx = 90, o_idx = 94`) without ever informing the user about it.
Imagine a user mistyped the `meta.json` file path. Because the program does not return any error upon not finding the file, the user will assume that the designated types specified in the `meta.json` file are being used. However, in reality, the hardcoded defaults are used and results returned by the program are for different inputs that the user made (or thought to have made).

My suggestion:
Because `meta.json` must be included for the analysis to work, let the program fail if the file is not found (ideally as early as possible). The user has to be informed that something that is required for the program to work properly is missing.
If all you need from the `meta.json` file in this function are `li_idx` and `o_idx`, consider if it makes sense to pass them as function arguments.

---

I agree. I was trying to use some part of the information that exists in the meta.json file (li_idx). But, unfortunately, we don't have the o_idx information in mata.json, and therefore that part should be hard coded. I will pass these variables to the function when I first define it. Then it's the user's responsibility to ensure the values are correct for future simulations when li_idx and o_idx are different.